dev_docs/contributing/kibana_http_api_design_guidelines.mdx

```diff
@@ -0,0 +1,613 @@
+---
+id: kibHttpApiGuidelines
+slug: /kibana-dev-docs/contributing/http-api-guidelines
+title: Guidelines for HTTP API design in Kibana
+description: Best practices and guidelines for designing consistent, user-friendly HTTP APIs in Kibana
+date: 2025-07-02
+tags: ['kibana', 'contributor', 'dev', 'http', 'api', 'rest', 'design']
+---
+
+[Click here](#structure-and-conventions) to skip to the structure and conventions guides.
+
+## Design principles
+
+### Consistency
+
+Consistency is key to a great API-user experience. APIs should be consistent in their naming, structure, and behavior. Follow established patterns when available. If no pattern exists, consider creating one paying close attention to:
+
+1. **Consistent naming**: Use the same suffix for IDs (e.g., `space_id`, `dashboard_id`, etc.). APIs with clear, descriptive field names help self-document how it should be used.
+2. **Consistent structure**: Use the same structure for referencing resources in requests and responses in all related APIs.
+3. **Consistent validation**: Apply the same validation rules across all APIs.
+4. **Consistent error messages**: Use the same error messages for the same errors.
```

Comment:
Are we speaking about identical errors & matching messages for different endpoints? For example, 

```
GET /api/foos/{id}
GET /api/foos
```

GET single object throws `404` with error message `not found`.
GET list of objects throws `404` with error message `not found`.

Reply:
I think we are talking more about the structure of error messages. I'll clarify this point and refer to later in the guide.